### PR TITLE
Add comments in exported variables and func in dep package

### PR DESCRIPTION
**Pull request by richardbertozzo**
Issue-id: #203

Description
Add comments in exported variables and functions for `dep` package.

### CODE REVIEW

File: dep/provider/authenticator.go [hunk 1]
```diff
@@ -7,8 +7,10 @@ import (
 	"github.com/byliuyang/app/fw"
 )
 
+// TokenValidDuration duration of a valid token.
```
Comment (byliuyang) [inline dep/provider/authenticator.go]:
```suggestion
// TokenValidDuration represents the duration of a valid token.
```

File: dep/provider/authenticator.go [hunk 2]
```diff
@@ -7,8 +7,10 @@ import (
...
 type TokenValidDuration time.Duration
 
+// Authenticator initializes Authenticator to generete and get information of tokens.
```
Comment (byliuyang) [inline dep/provider/authenticator.go]:
```suggestion
// Authenticator creates Authenticator with TokenValidDuration to uniquely identify duration during dependency injection
```

File: dep/provider/graphql.go [hunk 3]
```diff
@@ -6,8 +6,10 @@ import (
 	"github.com/byliuyang/app/fw"
 )
 
+// GraphQlPath GraphQl path.
```
Comment (byliuyang) [inline dep/provider/graphql.go]:
```suggestion
// GraphQlPath represents the path for GraphQl APIs.
```

File: dep/provider/graphql.go [hunk 4]
```diff
@@ -6,8 +6,10 @@ import (
...
 type GraphQlPath string
 
+// GraphGophers initializes a new GraphQl server.
```
Comment (byliuyang) [inline dep/provider/graphql.go]:
```suggestion
// GraphGophers creates GraphGopher GraphQL server with GraphQlPath to uniquely identify graphqlPath during dependency injection.
```

File: dep/provider/jwt.go [hunk 5]
```diff
@@ -6,8 +6,10 @@ import (
 	"github.com/byliuyang/app/fw"
 )
 
+// JwtSecret secret to encode and decode JWT.
```
Comment (byliuyang) [inline dep/provider/jwt.go]:
```suggestion
// JwtSecret represents the secret used to encode and decode JWT token.
```

File: dep/provider/jwt.go [hunk 6]
```diff
@@ -6,8 +6,10 @@ import (
...
 type JwtSecret string
 
+// JwtGo initializes Crypto Tokenizer for encode and decode tokens using the JWT secret.
```
Comment (byliuyang) [inline dep/provider/jwt.go]:
```suggestion
// JwtGo creates Crypto Tokenizer with JwtSecret to uniquely identify secret during dependency injection
```

File: dep/provider/oauth.go [hunk 7]
```diff
@@ -6,9 +6,13 @@ import (
 	"github.com/byliuyang/app/fw"
 )
 
+// GithubClientID Github client ID credential.
```
Comment (byliuyang) [inline dep/provider/oauth.go]:
```suggestion
// GithubClientID represents client ID used for Github OAuth
```

File: dep/provider/oauth.go [hunk 8]
```diff
@@ -6,9 +6,13 @@ import (
...
 type GithubClientID string
+
+// GithubClientSecret Github client secret credential.
```
Comment (byliuyang) [inline dep/provider/oauth.go]:
```suggestion
// GithubClientSecret represents client secret used for Github OAuth.
```

File: dep/provider/oauth.go [hunk 9]
```diff
@@ -6,9 +6,13 @@ import (
...
 type GithubClientSecret string
 
+// GithubOAuth create a new Github struct to get Authorization infos.
```
Comment (byliuyang) [inline dep/provider/oauth.go]:
```suggestion
// GithubOAuth creates a new Github OAuth client with GithubClientID and GithubClientSecret to uniquely identify clientID and clientSecret during dependency injection.
```

File: dep/provider/recaptcha.go [hunk 10]
```diff
@@ -7,8 +7,10 @@ import (
 	"github.com/byliuyang/app/fw"
 )
 
+// ReCaptchaSecret reCAPTCHA secret.
```
Comment (byliuyang) [inline dep/provider/recaptcha.go]:
```suggestion
// ReCaptchaSecret represents the secret used to verify reCAPTCHA.
```

File: dep/provider/recaptcha.go [hunk 11]
```diff
@@ -7,8 +7,10 @@ import (
...
 type ReCaptchaSecret string
 
+// ReCaptchaService initializes reCAPTCHA service.
```
Comment (byliuyang) [inline dep/provider/recaptcha.go]:
```suggestion
// ReCaptchaService creates reCAPTCHA service with ReCaptchaSecret to uniquely identify secret during dependency injection.
```

File: dep/provider/routes.go [hunk 12]
```diff
@@ -11,8 +11,10 @@ import (
 	"github.com/byliuyang/app/fw"
 )
 
+// WwwRoot the Root address service.
```
Comment (byliuyang) [inline dep/provider/routes.go]:
```suggestion
// WwwRoot represents the location of static assets for the web UI
```

File: dep/provider/routes.go [hunk 13]
```diff
@@ -11,8 +11,10 @@ import (
...
 type WwwRoot string
 
+// ShortRoutes initializes the service routes.
```
Comment (byliuyang) [inline dep/provider/routes.go]:
```suggestion
// ShortRoutes creates HTTP routes for Short API with WwwRoot to uniquely identify wwwRoot during dependency injection.
```

**Comment by byliuyang** [comment]
Hi @richardbertozzo, thank you for working this PR. Please address the comments.

**Comment by richardbertozzo** [comment]
Thanks @byliuyang for your code review and comments, i submitted the changes.

File: dep/provider/graphql.go [hunk 14]
```diff
@@ -6,8 +6,10 @@ import (
 	"github.com/byliuyang/app/fw"
 )
 
+// GraphQlPath represents the path for GraphQl APIs.
```
Comment (byliuyang) [inline dep/provider/graphql.go]:
```suggestion
// GraphQlPath represents the path for GraphQL APIs.
```

**Comment by byliuyang** [comment]
Hi @richardbertozzo, please address one last change. Other than that, this CL looks great!

**Comment by byliuyang** [comment]
Please do `Squash and merge`

**Comment by byliuyang** [comment]
https://github.com/byliuyang/short/issues/203

**Pull request merged**